apps/frontend/src/app/2_molecules/AmountRenderer/AmountRenderer.tsx

```diff
@@ -54,11 +57,32 @@ export const AmountRenderer: FC<AmountRendererProps> = ({
     [formattedValue, value],
   );
 
+  const countDecimalPlaces = useCallback((num, decimals) => {
```

Comment:
This method is completely unnecessary:
1. Extract a method for determining the decimal part length - create a helper method in `math.ts`.
2. Use this method here and round the value with `toFixed` if `number of decimals > decimalPrecision`.